src/deploy.ts

```diff
@@ -1,43 +1,133 @@
 import fs from 'fs'
 import util from 'util';
 import { ux, sdk } from '@cto.ai/sdk';
-import { exec as oexec } from 'child_process';
+import { exec as oexec, execSync } from 'child_process';
 const pexec = util.promisify(oexec);
-
+const ARGS = require('minimist')(process.argv.slice(2));
+import { stackEnvPrompt, stackRepoPrompt, stackTagPrompt } from './prompts';
 
 async function run() {
 
   const STACK_TYPE = process.env.STACK_TYPE || 'aws-ecs-fargate';
   const STACK_TEAM = process.env.OPS_TEAM_NAME || 'private'
 
-  await ux.print(`\n🛠 Loading the ${ux.colors.white(STACK_TYPE)} stack for the ${ux.colors.white(STACK_TEAM)}...\n`)
+  const { STACK_ENV } = await stackEnvPrompt()
+  const { STACK_REPO } = await stackRepoPrompt()
 
-  const { STACK_ENV } = await ux.prompt<{
-    STACK_ENV: string
-  }>({
-      type: 'input',
-      name: 'STACK_ENV',
-      default: 'dev',
-      message: 'What is the name of the environment?'
-    })
+  const ecrRepoName: string = `${STACK_REPO}-${STACK_TYPE}`
 
-  const { STACK_REPO } = await ux.prompt<{
-    STACK_REPO: string
-  }>({
-      type: 'input',
-      name: 'STACK_REPO',
-      default: 'sample-expressjs-aws-ecs-fargate',
-      message: 'What is the name of the application repo?'
+  // Validate if the AWS Creds are set.
+  try {
+    const accessKeyId = process.env.AWS_ACCESS_KEY_ID;
+    const secretAccessKey = process.env.AWS_SECRET_ACCESS_KEY;
+
+    if (accessKeyId && secretAccessKey) {
+      console.log('AWS credentials are set.');
+      // Proceed with the rest of the deployment logic
+    } else {
+      console.log('AWS credentials are not set.');
+      return;
+    }
+  } catch (error) {
+    console.error('Invalid credentials:', error);
+    return;
+  }
+
+  await ux.print(`\n🛠 Loading the latest tags for ${ux.colors.green(STACK_TYPE)} environment and ${ux.colors.green(STACK_REPO)} service...`)
+
+  async function retrieveCurrentlyDeployedImage(env: string, service: string): Promise<string> {
```

Comment:
Please correct me if I'm wrong. Why this function needs to be `async` (therefore return a `Promise<string>`)? I'm asking since I see that most of functions used here are `sync` and all the returned values are bare strings.